packages/cli/src/databases/migrations/postgresdb/1681134145996-AddUserActivatedProperty.ts

```diff
@@ -0,0 +1,59 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import { getTablePrefix, logMigrationEnd, logMigrationStart } from '@db/utils/migrationHelpers';
+import { userSettings } from '@/Interfaces';
+
+export class AddUserActivatedProperty1681134145996 implements MigrationInterface {
+	name = 'AddUserActivatedProperty1681134145996';
+
+	async up(queryRunner: QueryRunner): Promise<void> {
+		logMigrationStart(this.name);
+
+		const tablePrefix = getTablePrefix();
+
+		const activatedUsers: userSettings[] = await queryRunner.query(
+			`SELECT DISTINCT sw."userId" AS id,
+				JSONB_SET(COALESCE(u.settings::jsonb, '{}'), '{userActivated}', 'true', true) as settings
+			FROM  ${tablePrefix}workflow_statistics ws
+						JOIN ${tablePrefix}shared_workflow sw
+							ON ws."workflowId" = sw."workflowId"
+						JOIN ${tablePrefix}role r
+							ON r.id = sw."roleId"
+						JOIN "${tablePrefix}user" u
+							ON u.id = sw."userId"
+			WHERE ws.name = 'production_success'
+						AND r.name = 'owner'
+						AND r.scope = 'workflow'`,
+		);
+
+		const updatedUsers = activatedUsers.map((user) =>
+			queryRunner.query(
+				`UPDATE "${tablePrefix}user" SET settings = '${JSON.stringify(
+					user.settings,
+				)}' WHERE id = '${user.id}' `,
+			),
+		);
+
+		await Promise.all(updatedUsers);
+
+		if (!activatedUsers.length) {
+			await queryRunner.query(
+				`UPDATE "${tablePrefix}user" SET settings = JSONB_SET(COALESCE(settings::jsonb, '{}'), '{userActivated}', 'false', true)`,
+			);
+		} else {
+			const activatedUserIds = activatedUsers.map((user) => `'${user.id}'`).join(',');
+
+			await queryRunner.query(
+				`UPDATE "${tablePrefix}user" SET settings = JSONB_SET(COALESCE(settings::jsonb, '{}'), '{userActivated}', 'false', true) WHERE id NOT IN (${activatedUserIds})`,
+			);
+		}
+
+		logMigrationEnd(this.name);
+	}
+
+	async down(queryRunner: QueryRunner): Promise<void> {
+		const tablePrefix = getTablePrefix();
+		await queryRunner.query(
+			`UPDATE ${tablePrefix}user SET settings = JSON_EXTRACT_PATH(settings, 'userActivated')`,
```

Comment:
From [what I'm reading](https://www.postgresql.org/docs/9.3/functions-json.html), this extracts the value of the `userActivated` key from the `settings` JSON object. It doesn't remove it?